integrations/gitlab/gitlab_integration/events/hooks/project_files.py

```diff
@@ -0,0 +1,111 @@
+                "Invalid push hook. Missing one or more of the required fields (before, after, ref)"
+            )
+
+        resource_configs = typing.cast(
+            GitlabPortAppConfig, event.port_app_config
+        ).resources
+
+        matching_resource_configs = [
+            resource_config
+            for resource_config in resource_configs
+            if (
+                resource_config.kind == ObjectKind.FILE
+                and isinstance(resource_config, GitLabFilesResourceConfig)
+            )
+        ]
+        if not matching_resource_configs:
+            logger.info(
+                "Resource kind was not found in the config mapping, please update your config mapping to include the file kind"
+            )
+            return
+
+        for resource_config in matching_resource_configs:
+            selector = resource_config.selector
+            if not (selector.files and selector.files.path):
+                return
+
+            if self.gitlab_service.should_process_project(
+                gitlab_project, selector.files.repos
+            ):
+                matched_file_paths = self.gitlab_service._get_file_paths(
+                    gitlab_project, selector.files.path, gitlab_project.default_branch
+                )
+                await self._process_modified_files(
+                    gitlab_project, changed_files, matched_file_paths
+                )
+                await self._process_removed_files(
+                    gitlab_project, removed_files, selector.files.path
+                )
+
+    async def _process_modified_files(
+        self,
+        gitlab_project: Project,
+        modified_files: list[str],
+        matched_file_paths: list[str],
+    ) -> None:
+        """
+        Process the modified files and register them.
+        """
+        for modified_file in modified_files:
+            if modified_file in matched_file_paths:
+                file_data = self.gitlab_service.get_and_parse_single_file(
+                    gitlab_project, modified_file, gitlab_project.default_branch
+                )
+                if file_data:
+                    await ocean.register_raw(ObjectKind.FILE, [file_data])
+
+    async def _process_removed_files(
+        self, project: Project, removed_files: list[str], selector_path: str
+    ) -> None:
+        """
+        Process unregister the removed files.
+        """
+        for removed_file in removed_files:
+            ## after a file in GitLab, we can't get it's content using the project.files.get() api. The way to still delete the file from Port is to use the file_path. This will look for the file in the Port database and delete it according to the mapping with the repo.
+            if does_pattern_apply(selector_path, removed_file):
+                await ocean.unregister_raw(
+                    ObjectKind.FILE,
+                    [{"file": {"file_path": removed_file}, "repo": project.asdict()}],
+                )
```

Comment:
this will work as long as the identifier of the blueprint uses the file_path or a combination with the repo

```yaml
    port:
      entity:
        mappings:
          identifier: .repo.name + "-" + .file.file_path | gsub(" "; "")
          title: .repo.name + "-" + .file.file_path
          blueprint: '"gitlabFile"'
  ```
  
  or 
  
  ```yaml
     port:
      entity:
        mappings:
          identifier: .file.file_path
          title: .file.file_path
          blueprint: '"gitlabFile"' 
  ```

Reply:
I doubt that it will work, but if running `resync` it will remove the stale data right?

Reply:
yes running the resync will remove the stale data